Mailbox/Program.cs

```diff
@@ -89,17 +89,73 @@ static void Main(string[] args)
 
         public static string GetOwnersDisplay(Mailboxes mailboxes)
         {
-            
+            if (mailboxes is null)
+                throw new ArgumentNullException($"{nameof(mailboxes)} was found to be null.");
+
+            List<Person> distinctOwners = new List<Person>();
+
+            foreach (Mailbox mail in mailboxes)
+            {
+                if (!distinctOwners.Contains(mail.Owner))
+                    distinctOwners.Add(mail.Owner);
+            }
+
+            string result = "";
+            foreach(Person owner in distinctOwners)
+                result += $"{owner.ToString()}{Environment.NewLine}";
+
+            return result;
         }
 
         public static string GetMailboxDetails(Mailboxes mailboxes, int x, int y)
         {
-            
+            if (mailboxes is null)
+                throw new ArgumentNullException($"{nameof(mailboxes)} was found to be null.");
+
+            else if (x <= 0 || y <= 0)
+                throw new ArgumentException($"{nameof(x)}: {x}, {nameof(y)}: y. They both must be more than 0.");
+
+            string result = "";
+
+            foreach(Mailbox mail in mailboxes)
+            {
+                if (mail.Location == (x, y))
+                    result += mail.ToString() + Environment.NewLine;
+            }
+
+            if (result.Length == 0)
+                return null;
+
+            return result;
         }
 
-        public static Mailbox AddNewMailbox(Mailboxes mailboxes, string firstName, string lastName, Size size)
+        public static Mailbox AddNewMailbox(Mailboxes mailboxes, string firstName, string lastName, Sizes size)
         {
-            
+            if (mailboxes is null)
+                throw new ArgumentNullException($"{nameof(mailboxes)} was found to be null.");
```

Comment:
The parameter passed to this exception's constructor should be the parameter name; not a message
```suggestion
                throw new ArgumentNullException(nameof(mailboxes));
```